src/librustc_privacy/lib.rs

```diff
@@ -710,7 +710,7 @@ impl<'a, 'tcx> PrivacyVisitor<'a, 'tcx> {
             NamedField(name) => format!("field `{}` of {} is private",
                                         name, struct_desc),
             UnnamedField(idx) => format!("field #{} of {} is private",
```

Comment:
I'm inclined to agree about the error message itself -- it would probably look nicer just as "field `X` of Y is private". I figure that such a change would be better in another PR, but I could add it to this one.